homeassistant/components/inels/sensor.py

```diff
@@ -0,0 +1,159 @@
+"""Inelse sensor entity."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from operator import itemgetter
+from typing import Any
+
+from inelsmqtt.const import BATTERY, RFTI_10B, TEMP_IN, TEMP_OUT, TEMP_SENSOR_DATA
+from inelsmqtt.devices import Device
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, TEMP_CELSIUS, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .base_class import InelsBaseEntity
+from .const import COORDINATOR_LIST, DOMAIN, ICON_BATTERY, ICON_TEMPERATURE
+from .coordinator import InelsDeviceUpdateCoordinator
+
+
+@dataclass
+class InelsSensorEntityDescriptionMixin:
+    """Mixin keys."""
+
+    value: Callable[[Device], Any | None]
+
+
+@dataclass
+class InelsSensorEntityDescription(
+    SensorEntityDescription, InelsSensorEntityDescriptionMixin
+):
+    """Class for describing inels entities."""
+
+
+def _process_data(data: str, indexes: list) -> str:
+    """Process data for specific type of measurements."""
+    array = data.split("\n")[:-1]
+    data_range = itemgetter(*indexes)(array)
+    range_joined = "".join(data_range)
+
+    return f"0x{range_joined}"
+
+
+def __get_battery_level(device: Device) -> int | None:
+    """Get battery level of the device."""
+    if device.is_available is False:
+        return None
+
+    # then get calculate the battery. In our case iss 100 or 0
+    return (
+        100
+        if int(_process_data(device.state, TEMP_SENSOR_DATA[BATTERY]), 16) == 0
+        else 0
+    )
```

Comment:
If the battery level sensor is only `100` or `0` this is probably better as a binary_sensor